src/explainer.ts

```diff
@@ -1,30 +1,37 @@
 'use strict';
 
-import k8s = require('k8s');
+import request = require('request');
+import * as kubernetes from '@kubernetes/client-node';
 import * as pluralize from 'pluralize';
-import * as kubeconfig from './kubeconfig';
 import { formatComplex, formatOne, Typed, formatType } from "./schema-formatting";
 
 export function readSwagger(): Promise<any> {
-    return kubeconfig.readKubectlConfig().then((kc) => readSwaggerCore(kc));
-}
+    const kc = new kubernetes.KubeConfig();
+    kc.loadFromDefault();
 
-function fixUrl(kapi: any /* deliberately bypass type checks as .domain is internal */): void {
-    if (kapi.domain.endsWith('//')) {
-        kapi.domain = kapi.domain.substr(0, kapi.domain.length - 1);
-    }
+    return readSwaggerCore(kc);
 }
 
-function readSwaggerCore(kc: kubeconfig.KubeConfig): Promise<any> {
-    const k8sImpl: typeof k8s = require('k8s');
+function readSwaggerCore(kc: kubernetes.KubeConfig): Promise<any> {
+    const uri = `${kc.getCurrentCluster().server}/swagger.json`;
+    const opts: request.Options = {
+        url: uri,
+    };
+    kc.applyToRequest(opts);
+
     return new Promise((resolve, reject) => {
-        const kapi = k8sImpl.api(apiCredentials(kc));
-        fixUrl(kapi);
-        kapi.get('swagger.json', (err, data) => {
-            if (err) {
-                reject(err);
-            } else {
-                resolve(data);
+        request(uri, opts, (error, response, body) => {
+            if (error) {
+                reject(error);
```

Comment:
Do we need to `return` from the lambda after these `reject` calls?  I'm not very familiar with explicit promise construction, but these look like after calling `reject` they would fall through to the next statement...?

Reply:
(or of course use `else` to prevent fallthrough)

Reply:
good point, done!